packages/hop-node/src/watchers/ChallengeWatcher.ts

```diff
@@ -68,184 +76,137 @@ class ChallengeWatcher extends BaseWatcherWithEventHandlers {
+        `checking ${dbTransferRoots.length} challengeable root db items`
       )
     }
-  }
 
-  async checkChallengeFromDb () {
-    const dbTransferRoots = await this.db.transferRoots.getResolvableTransferRoots()
     for (const dbTransferRoot of dbTransferRoots) {
-      const {
-        sourceChainId,
-        destinationChainId,
-        transferRootHash,
-        totalAmount
-      } = dbTransferRoot
-      await this.checkChallenge(
-        sourceChainId,
-        destinationChainId,
-        transferRootHash,
-        totalAmount
+      const rootHash = dbTransferRoot.transferRootHash
+      if (this.shouldSkipChallenge[rootHash]) {
+        return
+      }
+
+      await this.checkChallengeableTransferRoot(
+        rootHash,
+        dbTransferRoot.totalAmount
       )
     }
   }
 
-  async checkTransferRoot (
+  async checkChallengeableTransferRoot (
     transferRootHash: string,
-    destinationChainId: number,
     totalAmount: BigNumber
   ) {
     const logger = this.logger.create({ root: transferRootHash })
-    logger.debug('handling L1 BondTransferRoot event')
+    const transferRootId = getTransferRootId(transferRootHash, totalAmount)
+
+    logger.debug('Challenging transfer root', transferRootHash)
     logger.debug('transferRootHash:', transferRootHash)
     logger.debug('totalAmount:', this.bridge.formatUnits(totalAmount))
-    logger.debug('destinationChainId:', destinationChainId)
+    logger.debug('transferRootId:', transferRootId)
 
-    if (isL1ChainId(destinationChainId)) {
-      // TODO
-      return
-    }
-
-    const l2Bridge = this.getSiblingWatcherByChainId(destinationChainId)
-      .bridge as L2Bridge
-    const blockNumber = await l2Bridge.getBlockNumber()
-    const recentTransferCommitEvents = await l2Bridge.getTransfersCommittedEvents(
-      blockNumber - 1000,
-      blockNumber
+    const dbTransferRoot = await this.db.transferRoots.getByTransferRootHash(
+      transferRootHash
     )
-    logger.debug('recent events:', recentTransferCommitEvents)
-
-    let found = false
-    for (let i = 0; i < recentTransferCommitEvents.length; i++) {
-      const { args, topics } = recentTransferCommitEvents[i]
-      const committedTransferRootHash = topics[1]
-      const committedTotalAmount = args[1]
-
-      if (
-        transferRootHash === committedTransferRootHash &&
-        totalAmount.eq(committedTotalAmount)
-      ) {
-        found = true
-        break
+
+    const currentChainId: number = await this.bridge.getChainId()
+    const activeChainIds: number[] = await this.bridge.getChainIds()
+    let isRootHashCommitted: boolean
+    let sourceChainId: number
+    for (const chainId of activeChainIds) {
+      if (currentChainId === chainId) {
+        continue
       }
-    }
 
-    if (found) {
-      logger.warn('transfer root committed')
-      return
+      const isRootHashCommitted = dbTransferRoot?.committed
+      if (isRootHashCommitted) {
+        sourceChainId = chainId
```

Comment:
Why is sourceChainId here assigned the first chainId that is not L1 in the list? Generally confused about this for loop

Reply:
Closing, as this is now stale.